src/java/org/apache/cassandra/io/sstable/Descriptor.java

```diff
@@ -145,7 +150,7 @@ public String filenameFor(Component component)
 
     public File fileFor(Component component)
     {
-        return component.getFile(baseFileURI);
+        return componentFileMap.computeIfAbsent(component, c -> component.getFile(baseFileURI));
```

Comment:
Was just curious about the nature of the race this is addressing - is it b/c the storage provider changes?